python/ray/dag/tests/test_torch_tensor_dag.py

```diff
@@ -0,0 +1,136 @@
+# coding: utf-8
+import logging
+import os
+import sys
+import torch
+
+import pytest
+
+import ray
+from ray.air._internal import torch_utils
+import ray.cluster_utils
+from ray.dag import InputNode
+from ray.tests.conftest import *  # noqa
+
+from ray.dag.experimental.types import TorchTensorType
+
+
+logger = logging.getLogger(__name__)
+
+if sys.platform != "linux" and sys.platform != "darwin":
+    pytest.skip("Skipping, requires Linux or Mac.", allow_module_level=True)
+
+
+@ray.remote
+class TorchTensorWorker:
+    def __init__(self):
+        self.device = torch_utils.get_devices()[0]
+
+    def send(self, shape, dtype, value: int):
+        return torch.ones(shape, dtype=dtype, device=self.device) * value
+
+    def recv(self, tensor):
+        # Check that tensor got loaded to the correct device.
+        assert tensor.device == self.device
+        return (tensor[0].item(), tensor.shape, tensor.dtype)
+
+
+@pytest.mark.parametrize("use_gpu", [False, True])
+def test_torch_tensor_p2p(ray_start_regular_shared, use_gpu):
```

Comment:
Does it make sense to split this test into multiple tests?

Reply:
I think it's okay either way, I just used the same test to reduce duplication.